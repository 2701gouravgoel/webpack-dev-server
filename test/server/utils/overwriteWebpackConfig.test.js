'use strict';

const overwriteWebpackConfig = require('../../../lib/utils/overwriteWebpackConfig');

describe('colors', () => {
  it('should set default color as true when config[0].stats is null', () => {
    const config = [{}];

    overwriteWebpackConfig(config);
    expect(config).toMatchSnapshot();
  });

  it('should pass the value when config[0].stats is a string', () => {
    const config = [{ stats: 'foo' }];

    overwriteWebpackConfig(config);
    expect(config).toMatchSnapshot();
  });

  it('should set default color as true when config[0].stats is an object', () => {
    const config = [{ stats: { foo: 'bar' } }];

    overwriteWebpackConfig(config);
    expect(config).toMatchSnapshot();
  });

  it('should not overwrite when config[0].stats.colors is not null', () => {
    const config = [{ stats: { colors: 'foo' } }];

    overwriteWebpackConfig(config);
    expect(config).toMatchSnapshot();
  });
});
