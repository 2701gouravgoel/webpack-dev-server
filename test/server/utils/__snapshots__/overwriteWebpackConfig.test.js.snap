// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colors should not overwrite when config[0].stats.colors is not null 1`] = `
Array [
  Object {
    "stats": Object {
      "colors": "foo",
    },
  },
]
`;

exports[`colors should pass the value when config[0].stats is a string 1`] = `
Array [
  Object {
    "stats": "foo",
  },
]
`;

exports[`colors should set default color as true when config[0].stats is an object 1`] = `
Array [
  Object {
    "stats": Object {
      "colors": true,
      "foo": "bar",
    },
  },
]
`;

exports[`colors should set default color as true when config[0].stats is null 1`] = `
Array [
  Object {
    "stats": Object {
      "colors": true,
    },
  },
]
`;
